matchbox/resource_group_test.go

```diff
@@ -64,3 +64,57 @@ func TestResourceGroup(t *testing.T) {
 	})
 
 }
+
+func TestResourceGroupJSON(t *testing.T) {
+	srv := NewFixtureServer(clientTLSInfo, serverTLSInfo, testfakes.NewFixedStore())
+	go srv.Start()
+	defer srv.Stop()
+
+	hcl := `
+		resource "matchbox_group" "default" {
+ 			name    = "default"
+  			profile = "foo"
+  			selector {
+				  qux = "baz"
+			}
+
+			metadata_json = "{\"foo\":\"bar\"}"
+		}
+	`
+
+	check := func(s *terraform.State) error {
+		grp, err := srv.Store.GroupGet("default")
+		if err != nil {
+			return err
+		}
+
+		if grp.GetId() != "default" {
+			return fmt.Errorf("id, found %q", grp.GetId())
+		}
+
+		if grp.GetProfile() != "foo" {
+			return fmt.Errorf("profile, found %q", grp.GetProfile())
+		}
+
+		selector := grp.GetSelector()
+		if len(selector) != 1 || selector["qux"] != "baz" {
+			return fmt.Errorf("selector.qux, found %q", selector["qux"])
+		}
+
+		if string(grp.GetMetadata()) != "{\"foo\":\"bar\"}" {
```

Comment:
Insted of copying the old tests, these could be written in the more readable style. Currently, if it fails, its not clear why.

```
fmt.Sprintf("expected metadata %v, got %v", expectedMetadata, group.GetMetadata())
```